packages/runtime/runtime-utils/src/summaryUtils.ts

```diff
@@ -252,6 +252,32 @@ export function convertSnapshotTreeToSummaryTree(
     return builder.getSummaryTree();
 }
 
+/**
+ * Utility to convert serialized snapshot taken in detached container to format where we can use it to
+ * attach the container.
+ * @param serializedSnapshotTree - serialized snapshot tree to be converted to summary tree for attach.
+ */
+export function convertSnapshotTreeToProtocolAndAppSummaryTree(
```

Comment:
Two observations:
1. Can we not expose internal details on API name? I.e. should host be aware of snapshot tree, protocol tree? It's conversion of container serialized format to driver serialized format - name should reflect that, not implementation details.
2. It's a bit concerning that we are created an API that is not called by Fluid repo itself. I.e. it suggests we have a copy / alternative way of generating what driver needs (when attachment goes through Container workflow) and that over time these two implementations are likely to diverge, i.e. we can easily break our consumers (nice to have UT, but even better would be for code to be used in prod).

Reply:
1.) Changed the api name to hide implementation details.
2.) Actually the input for both apis are different although the output is same. So I had to add a new api. As the api clearly suggests that the input is strictly the one taken by serialize() api for container.